draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +428,89 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+# Example Extensions
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
+Extensions can define new semantics for the payload of HTTP Datagrams. The
+extension can then have an endpoint pick an available context ID using the
+allocation service ({{context-id}}) and register that context ID with their
+peer. Registration is the action by which an endpoint informs its peer of the
+semantics and format of a given context ID.
 
 
---- back
+## Registering Contexts with Headers
+
+Extensions can define a new HTTP header to register a context ID with the peer
+endpoint.
+
+As an example, let's use an extension that conveys the time at which a UDP
+packet was received. The extension would first define the format of its HTTP
+Datagram Context Payload field:
+
+~~~
+Context Payload for UDP with Timestamp {
+  Timestamp (64),
+  UDP Payload (..),
+}
+~~~
+{: #ex-dgram title="Example: Format of UDP Payload with Timestamp"}
+
+The extension would also define a new HTTP header (UDP-Timestamps) that includes
+a context ID value. Endpoints that understand this new HTTP header would be able
+to consequently handle and parse datagrams on the context ID, while all other
+endpoints would ignore the datagrams.
```

Comment:
I've clarified the text about this specific client-initiated extension example.

However, we don't need to do anything else for unidirectional extensions. Such an extension could easily say that its registration is unidirectional, and that sending in the other direction is not allowed. There's no reason to split the namespace - QUIC did that because a bidirectional stream has a more complex state machine, and bidirectional flow control - neither of which apply to datagrams.